src/lib/permissions/interfaces.ts

```diff
@@ -1,57 +1,6 @@
-export type Resource = {
-  resourceId: string;
-};
-
-export type AssignablePermissionGroups = 'user' | 'role' | 'space' | 'public';
-
-// Public is a pure true / false. We only need to know there is a public group, to know it is true
-export type TargetPermissionGroup<G extends AssignablePermissionGroups = AssignablePermissionGroups> =
-  G extends 'public' ? { group: 'public' } : { group: G; id: string };
-
-export type PermissionAssignment<T extends AssignablePermissionGroups = AssignablePermissionGroups> = Resource & {
-  assignee: TargetPermissionGroup<T>;
-};
-
-/**
- * P left in as sometimes we want to specify all false / all true
- */
-export type UserPermissionFlags<T extends string, P extends boolean = boolean> = Record<T, P>;
-
-/**
- * This is the data we need to compute permissions for a resource
- * It is an improvement after Permission Compute Request, since we assume that all permission compute methods should be responsible for deciding if an admin override is available for a specific operation
- */
-export type PermissionCompute = {
-  resourceId: string;
-  userId?: string;
-};
-export type SpaceResourcesRequest = {
-  spaceId: string;
-  userId?: string;
-};
-
-/**
- * @publicOnly If true, ensures that authenticated requests will be treated as unauthenticated requests, only returning publicly available resources
- */
-export type AvailableResourcesRequest = SpaceResourcesRequest & {
-  publicOnly?: boolean;
-};
-
-export interface AbstractPermissions<O extends string> {
-  get empty(): UserPermissionFlags<O, false>;
-  get full(): UserPermissionFlags<O, true>;
-  get operationFlags(): UserPermissionFlags<O>;
-
-  addPermissions(operations: O[] | Partial<UserPermissionFlags<O>>): void;
-
-  hasPermissions(operations: O[]): boolean;
-}
-
-export type PermissionToDelete = {
```

Comment:
I liked the name `PermissionToDelete` better than `PermissionResource` because it is clear what it's for, and I don't see a current need to generalize it. I think you would need additional types for a resource if we are talking about the information to "updating" or "creating" it

Reply:
Ok can always change later since we have only the id needed for now for deletion, and that way we have a singular way of referring to a permissionId type resource